### ISSUE TITLE
Unable to retrieve the current column group definition

### ISSUE THREAD

**Issue by Heriannan** [ag-grid/ag-grid#6787]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```

**Current behavior**
ag-grid-react and ag-grid-community versions 30.0.0 always return an empty array when using gridApi.getColumnDefs() to retrieve the current column group definition. This issue is not present in previous versions such as 29.3.5. The problem may be caused by changes made in this file (https://github.com/ag-grid/ag-grid/blob/latest/grid-community-modules/core/src/ts/columns/columnDefFactory.ts) .

Reproducible scenario: https://plnkr.co/edit/K11KE9V3jCG5ltGC?open=index.tsx

**Expected behavior**
gridApi.getColumnDefs() should be able to retrieve the current column group definition.

**Please tell us about your environment:**

* **AG Grid version:**  30.0.0

* **Browser:** Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36

* **Language:** React

**Comment by braedonsaunders** [comment]
I have also noticed this, however the getColumnDefs is returning some columns for me - but not all. In this particular table, it is not returning any column with children:

This does appear:
`[
                        "headerName" => "Notes",
                        "field" => "memo",
                        "editable" => true,
                        "cellEditor" => "agLargeTextCellEditor",
                        "cellEditor" => "timesheetConditionalCellEditor",
                        'minWidth' => 70,
                        'maxWidth' => 150,
                        "cellEditorPopup" => true,
                        "cellEditorParams" => [
                            "maxLength" => 200,
                            "rows" => 10,
                            "cols" => 50
                        ],
                        'spanHeaderHeight' => true,
                        "cellClassRules" => [
                            "darker-cell" => [
                                "functionBody" => "return params.data && (params.data.department_timesheet_style === 'BillableTimesheet' && params.data.customer);"
                            ]
                        ]
                    ],`
                    
This column does not appear:
`[
                        'headerName' => 'SUN', 
                        'children' => [
                            [
                                "field" => "day_1",
                                'minWidth' => 80,
                                'maxWidth' => 80,
                                "editable" => true,
                                "cellEditor" => "timesheetConditionalCellEditor",
                                "headerValueGetter" => [
                                    "functionBody" => "return moment(document.getElementById('day1Header').value).format('MM-DD');"
                                ],
                                "cellClassRules" => [
                                    "darker-cell" => [
                                        "functionBody" => "return params.data && (params.data.department_timesheet_style === 'BillableTimesheet' && params.data.customer);"
                                    ]
                                ],
                                'aggFunc' => 'sum'
                            ],
                          ]
                    ],`

**Comment by kiril-matev** [comment]
Hello @Heriannan, @braedonsaunders,

Thank you both for reporting this - much appreciated!

Based on your description we've now reproduced this issue as a regression from v29.3.5.

We have added this bug to our backlog and we are tracking it with the following reference and description:
**AG-8813 [Regression] When calling api.getColumnDefs API method, column groups aren't returned**

We try to fix bugs from one release to the next, so this should be fixed in the next release or the one after if it was raised too close to the next release date.

See whether this item will be part of our next release by checking the NEXT RELEASE checkbox on the product pipeline page:
https://www.ag-grid.com/ag-grid-pipeline/

**Now that this issue is recorded in our backlog, we will soon close the GitHub issue because we will not update it when this bug is fixed in a future version of AG Grid. The best way to get an update on this item is to sign up for AG Grid new release notifications. Once a new AG Grid version is released, look for the item reference above (AG-XXXX) on the changelog page to see if it's resolved in that version:
https://www.ag-grid.com/ag-grid-changelog/**

Thanks again for bringing this up with us - we appreciate your time!
